packages/core/echo/echo-schema/src/typed-object.ts

```diff
@@ -348,6 +350,10 @@ class TypedObjectImpl<T> extends EchoObject<DocumentModel> {
       },
 
       set: (target, property, value, receiver) => {
+        if (this._opts?.readOnlyAccess) {
+          log.warn('Read only access');
+          return false;
```

Comment:
What would returning `false` do in this case? I think this should throw an error.

Also can you add a `isReadOnly` getter via a symbol key to the objects?